kubetest/kubeadmdind/kubeadm_dind.go

```diff
@@ -160,6 +160,29 @@ func (d *Deployer) outputWithStderr(cmd *exec.Cmd) ([]byte, error) {
 	return stdOutErr.Bytes(), err
 }
 
+func (_ *Deployer) GetKubectlVersion() error {
+	var (
+		interrupt = time.NewTimer(time.Duration(0)) // interrupt testing at this time.
+		terminate = time.NewTimer(time.Duration(0)) // terminate testing at this time.
+		verbose   = false
+		timeout   = time.Duration(0)
+		control   = process.NewControl(timeout, interrupt, terminate, verbose)
+		retries   = 5
+	)
+	for {
+		_, err := control.Output(exec.Command("./cluster/kubectl.sh", "--match-server-version=false", "version"))
+		if err == nil {
+			return nil
+		}
+		retries--
+		if retries == 0 {
+			return err
+		}
+		log.Print("Failed to reach api. Sleeping for 10 seconds before retrying...")
+		time.Sleep(10 * time.Second)
+	}
```

Comment:
this is duplicated logic.. maybe move the `getKubectlVersion` to util package so you can reuse it here?

Reply:
@krzyzacy Found out that would make `util` package import `process` package, thus cyclic imports. Do you have any other suggestion?

Reply:
we should probably break those up into more specific sub-packages

Reply:
@BenTheElder Can't think of better place to put `getKubectlVersion` other than `kubetest/e2e` package. Or maybe `kubetest/client`?